File: src/main/java/org/fletchly/genius/context/ContextService.java
package org.fletchly.genius.context;

import org.fletchly.genius.common.Message;
import org.fletchly.genius.context.model.ContextConversation;
import org.fletchly.genius.context.model.ContextMessage;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface ContextService {
    public CompletableFuture<ContextConversation> getOrCreateConversationForPlayer(UUID playerUuid);
    public CompletableFuture<Void> addMessageForPlayer(UUID playerUuid, Message message);
    public CompletableFuture<Void> deleteConversationForPlayer(UUID playerUuid);
}

File: src/main/kotlin/io/fletchly/genius/context/service/ContextService.kt
package io.fletchly.genius.context.service

import io.fletchly.genius.conversation.model.Message
import java.util.*

/**
 * Manages player context
 */
interface ContextService {
    /**
     * Add chat to context for given player and return updated context
     */
    suspend fun addChat(message: Message, playerUuid: UUID): List<Message>

    /**
     * Clear context for all players in the current session
     */
    suspend fun clearContext()

    /**
     * Clear context for specified player
     */
    suspend fun clearContext(playerUuid: UUID)
}